internal/componentsgen/utils.go

```diff
@@ -17,6 +18,35 @@ import (
 	securejoin "github.com/cyphar/filepath-securejoin"
 )
 
+func fetch(ctx context.Context, url, version, dir, filename string) ([]byte, error) {
```

Comment:
```suggestion
func fetchFromGhRelease(ctx context.Context, url, version, dir, filename string) ([]byte, error) {
```

because this is tied to github.